routing/graph.go

```diff
@@ -33,6 +33,59 @@ type ReadOnlyGraph interface {
 	Graph
 }
 
+type GraphSessionConstructor interface {
```

Comment:
could have the `Factory` suffix if that isn't too Java for you ;)

Reply:
updated :)